cmd/lakectl/cmd/import.go

```diff
@@ -50,99 +53,67 @@ var importCmd = &cobra.Command{
+			for sig := range c {
+				Fmt("\nCanceling import (reason: %s)\n", sig.String())
+				_, _ = client.ImportCancelWithResponse(ctx, toURI.Repository, toURI.Ref, &api.ImportCancelParams{Id: importID})
+				Die("Import Canceled", 1)
 			}
+		}()
 
-			continuationToken = rangeResp.JSON201.Pagination.ContinuationToken
-			after = rangeResp.JSON201.Pagination.LastKey
-			stagingToken = rangeResp.JSON201.Pagination.StagingToken
-			if !rangeResp.JSON201.Pagination.HasMore {
+		const StatusPollInterval = 10 * time.Second
+		statusResp, err := client.ImportStatusWithResponse(ctx, toURI.Repository, toURI.Ref, &api.ImportStatusParams{Id: importID})
+		for {
+			DieOnErrorOrUnexpectedStatusCode(statusResp, err, http.StatusOK)
+			status := statusResp.JSON200
+			if status.Error != nil {
+				DieFmt("Import failed: %s", status.Error.Message)
+			}
+			_ = bar.Set64(status.ImportProgress)
+			if status.Completed {
 				break
 			}
+			time.Sleep(StatusPollInterval)
+			statusResp, err = client.ImportStatusWithResponse(ctx, toURI.Repository, toURI.Ref, &api.ImportStatusParams{Id: importID})
 		}
 		_ = bar.Clear()
 
-		// create metarange with all the ranges we created
-		metaRangeResp, err := client.CreateMetaRangeWithResponse(ctx, toURI.Repository, api.CreateMetaRangeJSONRequestBody{
-			Ranges: ranges,
-		})
-		DieOnErrorOrUnexpectedStatusCode(metaRangeResp, err, http.StatusCreated)
-		if metaRangeResp.JSON201 == nil {
-			Die("Bad response from server", 1)
-		}
-
-		importedBranchID := formatImportedBranchID(toURI.Ref)
-		ensureBranchExists(ctx, client, toURI.Repository, importedBranchID, toURI.Ref)
-
-		// commit metarange to the imported branch
-		commitResp, err := client.CommitWithResponse(ctx, toURI.Repository, importedBranchID, &api.CommitParams{
-			SourceMetarange: metaRangeResp.JSON201.Id,
-		}, api.CommitJSONRequestBody{
-			Message: message,
-			Metadata: &api.CommitCreation_Metadata{
-				AdditionalProperties: metadata,
-			},
-		})
-		DieOnErrorOrUnexpectedStatusCode(commitResp, err, http.StatusCreated)
-		if commitResp.JSON201 == nil {
-			Die("Bad response from server", 1)
-		}
-
-		if stagingToken != nil && *stagingToken != "" {
-			stageResp, err := client.UpdateBranchTokenWithResponse(ctx, toURI.Repository, importedBranchID, api.UpdateBranchTokenJSONRequestBody{
-				StagingToken: *stagingToken,
-			})
-			DieOnErrorOrUnexpectedStatusCode(stageResp, err, http.StatusCreated)
-			// Commit staged data (skipped files)
-			commitResp, err = client.CommitWithResponse(ctx, toURI.Repository, importedBranchID, &api.CommitParams{}, api.CommitJSONRequestBody{
-				Message: "Import commit for staged objects",
-				Metadata: &api.CommitCreation_Metadata{
-					AdditionalProperties: metadata,
-				},
-			})
-			DieOnErrorOrUnexpectedStatusCode(commitResp, err, http.StatusCreated)
-			if commitResp.JSON201 == nil {
-				Die("Bad response from server", 1)
-			}
-		}
-
+		importedBranch := api.StringValue(statusResp.JSON200.ImportBranch)
```

Comment:
Both here and in the UI, I would suggest removing the intermediate branch. 
For the UI specifically, the import branch that is created has a wrong parent: instead of using the `Repository Created` commit as parent, it uses the same ref as the destination branch. 

This means if I run an import (from the UI) for the first time on a branch that contains data, **merging it will result in everything but the imported data being deleted**.

This is exactly the weird behaviors we wanted to avoid :)

Reply:
This PR does not contain all required changes for Import. In the PR description I've detailed a list of changes that are missing in this PR and should be implemented in follow-up issues, trying not to make this PR more complicated that it already is.
Note that the branch creation is done in the server now as part of the import API, so both the UI and lakectl flows are exactly the same.
This of course will be removed in a succeeding PR, once we understand how to do it best.